moto/ec2/responses/virtual_private_gateways.py

```diff
@@ -13,7 +13,14 @@ def attach_vpn_gateway(self):
 
     def create_vpn_gateway(self):
         type = self._get_param("Type")
-        vpn_gateway = self.ec2_backend.create_vpn_gateway(type)
+        amazon_side_asn = self._get_param("AmazonSideAsn")
+        availability_zone = self._get_param("AvailabilityZone")
+        tags = self._get_multi_param("TagSpecification")
```

Comment:
nit: Looks like we're using the same code for extracting tags (lines 18-21 in this file) in quite a few places - perhaps we can extract it into a small util function, to reduce duplication?

Reply:
Sounds good.

Reply:
@monty16597 I guess we could use the new `add_tag_specification()` util function here.